lib/puppet/transaction/resource_harness.rb

```diff
@@ -107,10 +107,10 @@ def persist_system_values(resource, context)
     end
 
     context.system_value_params.each do |pname, param|
-      @persistence.set_system_value(resource.name, pname.to_s,
+      @persistence.set_system_value(resource.to_s, pname.to_s,
```

Comment:
Nit, we typically use `resource.ref` to lookup resources, eg. when autorequiring, etc, but `ref` just calls `to_s`, so what you have is correct.